### PR TITLE
MAKE-589: Add clone and dry run to s3 impl

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

**Comment by tychoish** [comment]
I think we can get away without the clone function, just the dryrun operation?

File: interface.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ type Bucket interface {
 	// List provides a way to iterator over the contents of a
 	// bucket (for a given prefix.)
 	List(context.Context, string) (BucketIterator, error)
+
+	// Copy bucket contents into a new bucket, with the option of setting
+	// dryRun to true.
+	Clone(bool) Bucket
```
Comment (tychoish) [inline interface.go]: I'm not sure if I like this API.

Comment (julianedwards) [reply interface.go]: Is it the name or the idea of cloning?

Comment (julianedwards) [reply interface.go]: Ok removed clone

**Pull request merged**